packages/debugger/src/sidebar.ts

```diff
@@ -79,16 +79,48 @@ export class DebuggerSidebar extends Panel {
       header.title.label = title;
     });
 
-    const body = new SplitPanel();
+    this.body = new SplitPanel();
+    this.body.orientation = 'vertical';
+    this.body.addClass('jp-DebuggerSidebar-body');
+    this.addWidget(this.body);
 
-    body.orientation = 'vertical';
-    body.addWidget(this.variables);
-    body.addWidget(this.callstack);
-    body.addWidget(this.breakpoints);
-    body.addWidget(this.sources);
-    body.addClass('jp-DebuggerSidebar-body');
+    this.addPanel(this.variables);
+    this.addPanel(this.callstack);
+    this.addPanel(this.breakpoints);
+    this.addPanel(this.sources);
+  }
 
-    this.addWidget(body);
+  /**
+   * Add a panel at the end of the sidebar.
+   *
+   * @param panel - The panel to add to the sidebar.
+   *
+   * #### Notes
+   * If the widget is already contained in the sidebar, it will be moved.
+   */
+  addPanel(panel: Panel) {
```

Comment:
Should the interface be restricted to `Panel` objects, or be extended to `Widget`?

Reply:
I think `addWidget(widget: Widget);` makes sense.

I wonder if it should return an `IDisposable` object to remove it from the panel or not, but that can be added later if we find it is helpful.

Reply:
Sounds reasonable will make that change.

Reply:
Seems renaming `addPanel` to `addWidget` and `insertPanel` to `insertWidget` would override methods of `Panel` which  `DebuggerSidebar` extends. I can do that but I am not sure if there will be any unforcen side effects.
Should I just leave the name and just change the signature? (ie: `addPanel(widget: Widget)` )
Any thoughts?

Reply:
Do you think it would be useful to require that  the Sidebar panels conform to a certain interface? So instead of a widget we would have a custom interface.

Reply:
Looks like `addPanel(widget: Widget)` would work too yes.

Reply:
> I wonder if it should return an IDisposable object to remove it from the panel or not, but that can be added later if we find it is helpful.

For now setting `widget.parent = null` should already make it possible to remove the widget from the panel.